test/integration/deployment/util.go

```diff
@@ -0,0 +1,330 @@
+	_, s, closeFn := framework.RunAMaster(masterConfig)
+
+	config := restclient.Config{Host: s.URL}
+	clientSet, err := clientset.NewForConfig(&config)
+	if err != nil {
+		t.Fatalf("error in create clientset: %v", err)
+	}
+	resyncPeriod := 12 * time.Hour
+	informers := informers.NewSharedInformerFactory(clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "deployment-informers")), resyncPeriod)
+
+	dc := deployment.NewDeploymentController(
+		informers.Extensions().V1beta1().Deployments(),
+		informers.Extensions().V1beta1().ReplicaSets(),
+		informers.Core().V1().Pods(),
+		clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "deployment-controller")),
+	)
+	rm := replicaset.NewReplicaSetController(
+		informers.Extensions().V1beta1().ReplicaSets(),
+		informers.Core().V1().Pods(),
+		clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "replicaset-controller")),
+		replicaset.BurstReplicas,
+	)
+
+	if err != nil {
+		t.Fatalf("Failed to create deployment controller")
+	}
+	return s, closeFn, rm, dc, informers, clientSet
+}
+
+func addPodConditionReady(pod *v1.Pod, time metav1.Time) {
+	pod.Status = v1.PodStatus{
+		Phase: v1.PodRunning,
+		Conditions: []v1.PodCondition{
+			{
+				Type:               v1.PodReady,
+				Status:             v1.ConditionTrue,
+				LastTransitionTime: time,
+			},
+		},
+	}
+}
+
+func (d *deploymentTester) logReplicaSetsOfDeployment(allOldRSs []*v1beta1.ReplicaSet, newRS *v1beta1.ReplicaSet) {
+	if newRS != nil {
+		d.t.Logf("New ReplicaSet of Deployment %s:\n%+v", d.deployment.Name, *newRS)
+	} else {
+		d.t.Logf("New ReplicaSet of Deployment %s is nil.", d.deployment.Name)
+	}
+	if len(allOldRSs) > 0 {
+		d.t.Logf("All old ReplicaSets of Deployment %s:", d.deployment.Name)
+	}
+	for i := range allOldRSs {
+		d.t.Logf(spew.Sprintf("%#v", *allOldRSs[i]))
+	}
+}
+
+func (d *deploymentTester) logPodsOfDeployment(rsList []*v1beta1.ReplicaSet) {
+	minReadySeconds := d.deployment.Spec.MinReadySeconds
+	podListFunc := func(namespace string, options metav1.ListOptions) (*v1.PodList, error) {
+		return d.c.Core().Pods(namespace).List(options)
+	}
+
+	podList, err := deploymentutil.ListPods(d.deployment, rsList, podListFunc)
+
+	if err != nil {
+		d.t.Logf("Failed to list Pods of Deployment %s: %v", d.deployment.Name, err)
+		return
+	}
+	for _, pod := range podList.Items {
+		availability := "not available"
+		if podutil.IsPodAvailable(&pod, minReadySeconds, metav1.Now()) {
+			availability = "available"
+		}
+		d.t.Logf("Pod %s is %s:\n%s", pod.Name, availability, spew.Sprintf("%#v", pod))
+	}
+}
+
+// WaitForDeploymentRevisionAndImage waits for the deployment's and its new RS's revision and container image to match the given revision and image.
+// Note that deployment revision and its new RS revision should be updated shortly, so we only wait for 1 minute here to fail early.
+func (d *deploymentTester) waitForDeploymentRevisionAndImage(revision, image string) error {
```

Comment:
There is a lot of overlap between these and the e2e utilities - can we share code?